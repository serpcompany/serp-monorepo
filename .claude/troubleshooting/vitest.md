https://vitest.dev/guide/common-errors.html