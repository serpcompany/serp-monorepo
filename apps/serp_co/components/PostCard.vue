<template>
  <article class="py-16">
    <nuxt-link :to="`/posts/${post.slug}/`">
      <nuxt-img
        v-if="post.image"
        :src="post.image"
        :alt="post.title"
        class="mb-6 h-64 w-full object-cover"
      />
      <!-- Title -->
      <h2 class="mb-2 text-xl hover:underline">
        {{ post.title }}
      </h2>
    </nuxt-link>

    <p class="mb-2 italic">
      <span>By {{ post.author }}</span
      ><span v-if="post.createdAt"> | {{ post.createdAt }}</span>
    </p>
    <p class="mb-8">{{ post.excerpt }}</p>

    <s-pill
      v-if="post.categories && post.categories.length"
      base-slug="posts/category"
      :items="post.categories"
    />
  </article>
</template>

<script setup lang="ts">
import type { PostIndex } from '~/types';

defineProps<{
  post: PostIndex;
}>();
</script>
