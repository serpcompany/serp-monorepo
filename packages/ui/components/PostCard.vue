<template>
  <article :class="articleClass">
    <nuxt-link :to="`/${baseSlug}${post.slug}/`">
      <lazy-nuxt-img
        v-if="post.featuredImage"
        :src="post.featuredImage"
        :alt="displayTitle"
        class="mb-6 h-64 w-full object-cover"
      />
      <!-- Title -->
      <h2 class="mb-2 text-xl font-medium hover:underline">
        {{ displayTitle }}
      </h2>
    </nuxt-link>

    <p v-if="post.author" class="mb-2 italic">
      <span>By {{ post.author }}</span
      ><span v-if="post.createdAt"> | {{ post.createdAt }}</span>
    </p>
    <p v-if="post.excerpt" class="mb-8">{{ post.excerpt }}</p>

    <s-pill
      v-if="post.categories && post.categories.length"
      :base-slug="`${baseSlug}category`"
      :items="post.categories"
    />
  </article>
</template>

<script setup lang="ts">
import type { PostIndex } from '@serp/types/types';

const props = withDefaults(
  defineProps<{
    post: PostIndex;
    baseSlug?: string;
    articleClass?: string;
    title?: string;
  }>(),
  {
    baseSlug: 'posts/',
    articleClass: 'py-16',
    title: undefined
  }
);

const displayTitle = computed(() => {
  if (props.post.module === 'Glossary') {
    return props.post.keyword || props.post.title;
  }
  if (props.title) {
    return props.title;
  }
  return props.post.title;
});
</script>
