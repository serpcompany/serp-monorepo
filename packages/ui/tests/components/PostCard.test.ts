import { describe, expect, it } from 'vitest';
import PostCard from '../../components/PostCard.vue';
import ComponentRender from '../componentRender';

describe('PostCard Snapshot', () => {
  // Define test scenarios covering various valid prop combinations
  const scenarios: [string, { props: unknown }][] = [
    [
      'with complete post data',
      {
        props: {
          post: {
            title: 'Test Title',
            module: 'Article',
            featuredImage: 'https://example.com/image.jpg',
            author: 'John Doe',
            createdAt: '2022-10-01',
            excerpt: 'This is a test excerpt for the post.',
            categories: [{ id: 1, slug: 'tech', name: 'Tech' }]
          },
          baseSlug: 'posts/',
          articleClass: 'py-16'
        }
      }
    ],
    [
      'without featured image and author',
      {
        props: {
          post: {
            title: 'No Image Post',
            module: 'Article',
            featuredImage: '',
            excerpt: 'Post without featured image and author info.',
            categories: []
          },
          baseSlug: 'posts/',
          articleClass: 'py-16'
        }
      }
    ],
    [
      'with custom title prop overriding display title',
      {
        props: {
          post: {
            title: 'Default Title',
            module: 'Article',
            featuredImage: 'https://example.com/image2.jpg',
            author: 'Jane Doe',
            createdAt: '2023-03-01',
            excerpt: 'Post with custom title prop.',
            categories: [{ id: 2, slug: 'news', name: 'News' }]
          },
          baseSlug: 'posts/',
          articleClass: 'py-16',
          title: 'Custom Title'
        }
      }
    ]
  ];

  it.each(scenarios)(
    'renders %s correctly',
    async (desc: string, options: { props: unknown }) => {
      const html = await ComponentRender(`PostCard ${desc}`, options, PostCard);
      expect(html).toMatchSnapshot();
    }
  );
});
