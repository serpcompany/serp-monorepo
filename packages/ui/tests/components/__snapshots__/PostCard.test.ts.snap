// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`postCard Snapshot > renders with complete post data correctly 1`] = `
<article class="
  gap-6
  grid
  grid-cols-1
  lg:grid-cols-12
  py-16
">
  <div class="lg:col-span-8">
    <a href="/posts/undefined/"><img
      alt="Test Title"
      class="
        h-64
        mb-6
        object-cover
        w-full
      "
      data-nuxt-img=""
      src="https://example.com/image.jpg"
      srcset="https://example.com/image.jpg 1x, https://example.com/image.jpg 2x"
    />
    <!-- Title -->
      <h2 class="
      font-extrabold
      hover:underline
      mb-4
      text-4xl
    ">Test Title</h2>
    </a>
    <p class="
      mb-4
      text-xl
    ">
      <span>
        This is a test excerpt for the post.
      </span>
    </p>
    <div class="mb-4">
      <span class="
        font-medium
        mr-1
        text-sm
      ">
        OCT 01 2022
      </span>
      <span>
        –
      </span>
      <span class="
        font-medium
        ml-1
        text-sm
      ">
        JOHN DOE
      </span>
    </div>
    <div class="
      flex
      flex-wrap
      gap-2
      mt-4
    ">
      <a
        class="
          bg-blue-50
          border
          border-blue-300
          dark:bg-blue-900/30
          dark:border-blue-700
          dark:hover:bg-blue-800/50
          dark:text-blue-300
          font-medium
          hover:bg-blue-100
          px-3
          py-1
          rounded-full
          text-blue-600
          text-sm
        "
        href="/posts/category/tech/"
      >Tech</a>
    </div>
  </div>
  <div class="
    lg:col-span-3
    lg:col-start-8
  ">
    <!-- Right column (30%) -->
  </div>
</article>
`;

exports[`postCard Snapshot > renders with custom title prop overriding display title correctly 1`] = `
<article class="
  gap-6
  grid
  grid-cols-1
  lg:grid-cols-12
  py-16
">
  <div class="lg:col-span-8">
    <a href="/posts/undefined/"><img
      alt="Custom Title"
      class="
        h-64
        mb-6
        object-cover
        w-full
      "
      data-nuxt-img=""
      src="https://example.com/image2.jpg"
      srcset="https://example.com/image2.jpg 1x, https://example.com/image2.jpg 2x"
    />
    <!-- Title -->
      <h2 class="
      font-extrabold
      hover:underline
      mb-4
      text-4xl
    ">Custom Title</h2>
    </a>
    <p class="
      mb-4
      text-xl
    ">
      <span>
        Post with custom title prop.
      </span>
    </p>
    <div class="mb-4">
      <span class="
        font-medium
        mr-1
        text-sm
      ">
        MAR 01 2023
      </span>
      <span>
        –
      </span>
      <span class="
        font-medium
        ml-1
        text-sm
      ">
        JANE DOE
      </span>
    </div>
    <div class="
      flex
      flex-wrap
      gap-2
      mt-4
    ">
      <a
        class="
          bg-blue-50
          border
          border-blue-300
          dark:bg-blue-900/30
          dark:border-blue-700
          dark:hover:bg-blue-800/50
          dark:text-blue-300
          font-medium
          hover:bg-blue-100
          px-3
          py-1
          rounded-full
          text-blue-600
          text-sm
        "
        href="/posts/category/news/"
      >News</a>
    </div>
  </div>
  <div class="
    lg:col-span-3
    lg:col-start-8
  ">
    <!-- Right column (30%) -->
  </div>
</article>
`;

exports[`postCard Snapshot > renders without featured image and author correctly 1`] = `
<article class="
  gap-6
  grid
  grid-cols-1
  lg:grid-cols-12
  py-16
">
  <div class="lg:col-span-8">
    <a href="/posts/undefined/">
      
    <!-- v-if -->
      
    <!-- Title -->
      <h2 class="
      font-extrabold
      hover:underline
      mb-4
      text-4xl
    ">No Image Post</h2>
    </a>
    <p class="
      mb-4
      text-xl
    ">
      <span>
        Post without featured image and author info.
      </span>
    </p>
    <!-- v-if -->
    <!-- v-if -->
  </div>
  <div class="
    lg:col-span-3
    lg:col-start-8
  ">
    <!-- Right column (30%) -->
  </div>
</article>
`;
